app/scripts/lib/transaction/smart-transactions.ts

```diff
@@ -33,19 +32,21 @@ import {
 
 const namespace = 'SmartTransactions';
 
-type AllowedActions =
+export type AllowedActions =
   | AddApprovalRequest
   | UpdateRequestState
   | StartFlow
   | EndFlow;
 
+export type AllowedEvents = SmartTransactionsControllerSmartTransactionEvent;
+
 export type SmartTransactionsControllerMessenger =
   RestrictedControllerMessenger<
     typeof namespace,
     AllowedActions,
-    EventConstraint,
+    AllowedEvents,
```

Comment:
Shouldn't we align these types with the allowlists used to instantiate `smartTransactionsControllerMessenger` in `MetamaskController`? (are the current `AllowedActions` used by the messenger?)

Also, is there a reason we can't include the `getState` action and `stateChange` event in these types?

```suggestion
export type AllowedActions = NetworkControllerGetNetworkClientByIdAction;

export type AllowedEvents = NetworkControllerStateChangeEvent;

export type SmartTransactionsControllerMessenger =
  RestrictedControllerMessenger<
    typeof namespace,
    SmartTransactionsControllerActions | AllowedActions,
    SmartTransactionsControllerEvents | AllowedEvents,
```

Reply:
The name `SmartTransactionsControllerMessenger` is misleading because it suggests that it's the same messenger used by `SmartTransactionsController`, which isn't the case. In fact, the `SmartTransactionHook` file takes a `controllerMessenger` that includes all previously defined and authorized `AllowedActions` from the account controller. However, within the current scope of work, this hook now requires `SmartTransactionsControllerSmartTransactionEvent` to subscribe to smartTransaction events.

Reply:
I just did rename to `SmartTransactionHookMessenger` to avoid confusion